test/run_envoy_bazel_coverage.sh

```diff
@@ -118,9 +118,9 @@ fi
 set +e
 if [[ "$VALIDATE_COVERAGE" == "true" ]] && [[ "${FUZZ_COVERAGE}" == "false" ]]; then
   echo "Checking per-extension coverage"
-  output=$(./test/per_file_coverage.sh)
+  output=$(sh ./test/per_file_coverage.sh)
```

Comment:
i think we still need to remove the `sh ` - there is bash (eg `[[`) syntax in the target file

Reply:
```
$ sh test/per_file_coverage.sh 
test/per_file_coverage.sh: 5: Syntax error: "(" unexpected
```

Reply:
sh test/per_file_coverage.sh  runs for me and I prefer better handling of -x :-(

Reply:
wierd - for ref this is my shell

```console
$ $SHELL --version
GNU bash, version 5.2.2(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

Reply:
> sh test/per_file_coverage.sh runs for me and I prefer better handling of -x :-(

not sure iiuc - but this file cannot run with actual `sh` it has incorrect syntax - im guessing if it works then `sh` is being aliased to bash or something that implements bash syntax (my `sh` is really `dash`)